components/CreatePetForm/CreatePetForm.js

```diff
@@ -82,13 +83,35 @@ const StyledCharacteristicsContainer = styled.div`
   }
 `;
 
-export default function CreatePetForm({ onCreatePet }) {
+export default function CreatePetForm({
+  initialData,
+  onCreatePet,
+  buttonLabel = "Create",
+  hideButtons = false,
+  onUpdatePet,
+}) {
   const [currentPet, setCurrentPet] = useState(0);
   const [characteristics, setCharacteristics] = useState({
     characteristic1: "",
     characteristic2: "",
   });
 
+  const [petName, setPetName] = useState("");
+
+  useEffect(() => {
+    if (initialData) {
+      const petIndex = animalList.findIndex(
+        (animal) => animal.type === initialData.type
+      );
```

Comment:
Füge ein `Error handling` ein, falls die initialData nicht die erwarteten Eigenschaften hat 

Überprüft den `petIndex` 
```
if (petIndex === -1) {
console.error("Der Typ in initialData wurde nicht in der animalList gefunden");
return;

Wenn der petIndex -1 ist bedeutet das, dass der type von initialData nicht in animalList gefunden wurde. In diesem Fall würdet ihr in der Konsole eine Fehlermeldung erhalten und die Funktion würde beendet werden. 
    }
```

Überprüft die `characteristics` 

```
if (!Array.isArray(initialData.characteristics) ||
initialData.characteristics.length < 2) {
console.error("Die Eigenschaften in initialData sollten ein Array mit mindestens zwei Elementen sein.");
 return;
}
```

Stellt mit dem Code sicher, dass initialData.characteristics ein Array ist und mindestens 2 Elemente enthält. Wenn das nicht der Fall ist, bekommt ihr eine Fehlermeldung in der Konsole zurück und die Funktion wird beendet. 

Überprüft `initialData.name`

Stellt sicher, dass der `petName` existiert. Falls er nicht definiert ist müsst ihr petName als leeren `string` setzen